### DISCUSSION TITLE
How to efficiently plot irregular volumes in Pyvista?

### DISCUSSION THREAD

**Discussion by NicoDK203** [pyvista/pyvista#5806] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import numpy as np
import pyvista as pv
import pyvistaqt as pvqt

points_matrix = np.array([[6.1266, 5.2204, 5],
                          [12.5, 30, 0],
                          [18.6266, 25.2204, 5],
                          [12.2532, 28.2578, 10],
                          [6.1266, 33.0373, 5],
                          [18.6266, 23.0373, 5],
                          [19.1935, 17.8462, 20],
                          [13.0669, 22.6257, 15],
                          [14.0136, 25.6613, 15],
                          [16.4666, 17.4916, 20],
                          [17.9571, 20.8818, 20],
                          [11.8305, 25.6613, 15]])

# Define combinations matrix (3 x 6)
combinations = np.array([
    [0, 1, 2, 3, 4, 5],
    [6, 7, 8, 9, 10, 11]
])

# Create PolyData objects for each volume and generate outer surfaces
outer_surfaces = [pv.PolyData(points_matrix[combo, :]).delaunay_3d() for combo in combinations]

# Plot all outer surfaces
plotter = pvqt.BackgroundPlotter()
for surface in outer_surfaces:
    plotter.add_mesh(surface, show_edges=True)
plotter.show()
```

### Description

I am trying to process and plot some data that is returned from a 3D modelling software. I managed to extract the data from the output files and set it in an understandable format. The points_matrix is an N x 3 matrix of N points with X, Y and Z coordinates. The volumes are made up by a combination of six of those points, which can be found in the M x 6 matrix.

I have tried plotting these volumes using pyvista's celltypes, but I don't think any are applicable for my case. The example I have given uses the inbuilt Delaunay3d function which properly plots the volumes but does so too slow for larger datasets (due to the for-loop and probably the slow delaunay function.

Is there any other way that I am not yet familiar with in pyvista which can handle plotting these volumes?

### System Information

```shell
(No bugs in the code)
```

### Screenshots

![image](https://github.com/pyvista/pyvista/assets/164203635/1b697130-8475-4f0b-ba1c-db76d02d38ae)

**Comment by MatthewFlamm** [comment]
The best method would be to determine what cell type these are and then create the mesh without using `delaunay_3d`.  6 points that define a single 3D cell would be a `vtkWedge`, however the examples you've shared do not look like a Wedge.  It looks like it is probably two cells that are combined together, e.g. two pyramids that share a quadrilateral face.
